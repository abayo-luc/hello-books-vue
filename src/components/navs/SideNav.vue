<template>
  <div class="side-nav">
    <div class="pages">
      <router-link v-for="(page, index) in pages" :key="index" :to="page.path">
        <div class="single-item" :class="{active: page.active}">
          <img :src="page.icon" class="icon" />
          <span>{{page.name}}</span>
        </div>
      </router-link>
    </div>
  </div>
</template>
<script>
import Icons from '../../assets/icons/index';

export default {
  name: 'SideNav',
  data() {
    return {
      pages: [
        {
          path: '/',
          name: 'Home',
          icon: Icons.Home
        },
        {
          path: '/search',
          name: 'Browse',
          icon: Icons.Search
        },
        {
          path: '/authors',
          name: 'Authors',
          icon: Icons.Avatar
        }
      ]
    };
  },
  created() {
    const currentPage = this.pages.find(page => page.path === this.$route.path);
    currentPage.active = true;
  },
  watch: {
    $route(to, from) {
      this.pages = this.pages.map((page) => {
        switch (page.path) {
          case to.path:
            return { ...page, active: true };
          case from.path:
            return { ...page, active: false };
          default:
            return page;
        }
      });
    }
  }
};
</script>
<style lang="scss" scoped>
.side-nav {
  margin: 2em;
  margin-top: 5em;
  a {
    color: inherit;
    text-decoration: none;
  }
  @media (max-width: 1024px) {
    display: none;
  }
}
.single-item {
  display: flex;
  flex-direction: row;
  align-items: center;
  &.active {
    span {
      border-bottom: 1px solid rgb(35, 33, 38);
      padding-bottom: 3px;
    }
  }
  img {
    margin: 10px;
  }
}
</style>
