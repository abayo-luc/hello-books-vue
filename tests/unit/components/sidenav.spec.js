import {
  mount
} from '@vue/test-utils';
import SideNav from '../../../src/components/navs/SideNav.vue';

const $route = {
  path: '/'
};
const wrapper = mount(SideNav, {
  mocks: {
    $route
  },
  stubs: ['router-link', 'router-view']
});

describe('SideNav.vue', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should container all pages', () => {
    expect(wrapper.vm.$data.pages.length > 2).toBeTruthy();
    expect(wrapper.vm.$data.pages[0].active).toBeTruthy();
  });
});
