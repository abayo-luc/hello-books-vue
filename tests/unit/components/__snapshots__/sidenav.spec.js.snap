// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav.vue should match snapshot 1`] = `
<div class="side-nav">
  <div class="pages">
    <router-link-stub to="/">
      <div class="single-item active"><img src="" class="icon"> <span>Home</span></div>
    </router-link-stub>
    <router-link-stub to="/search">
      <div class="single-item"><img src="" class="icon"> <span>Browse</span></div>
    </router-link-stub>
    <router-link-stub to="/authors">
      <div class="single-item"><img src="" class="icon"> <span>Authors</span></div>
    </router-link-stub>
  </div>
</div>
`;
